HACKING.md

```diff
@@ -141,17 +141,82 @@ appropriately.  To install `godep` locally run:
 
 If you are not updating packages you should not need godep installed.
 
-## Updating Godeps from upstream
+## Updating Kubernetes from upstream
+
+There are a few steps involved in rebasing Origin to a new version of Kubernetes. We need to make sure
```

Comment:
Besides the script @smarterclayton mentioned before rebasing k8s is invoking `godep restore` inside origin (https://github.com/openshift/origin/blob/master/hack/rebase-kube.sh#L30-L32). This is needed, because origin has some dependencies which do not exist in k8s, eg. STI (https://github.com/openshift/origin/blob/master/Godeps/Godeps.json#L504-L506). Your below description does not cover that, that means that invoking `godep save` will erase such deps. It'll be worth mentioning that you should check the diff of `Godeps.json` file if it's not missing any deps and if so check if that was intentional.